JavaScript/es6Guidelines.md

```diff
@@ -0,0 +1,204 @@
+
+    this cannot be changed inside the function
+
+3.  **Implied Returns**
+
+    Curly braces are optional for arrow functions, they can be left off to create a super simple function. In those types of definitions, the result of the expression is always returned.
+
+    Example:
+
+    ```javascript
+    let sum = (x, y) => x + y;
+    ```
+
+
+4.  **`arguments` gotcha**
+
+    Check out following example: (code comes from [Ooyala Player](https://github.com/Wikia/mercury/blob/dev/front/scripts/mercury/modules/VideoPlayers/Ooyala.js))
+    ```javascript
+    setupPlayer() {
+      this.params = $.extend(this.params, {
+        onCreate: (...args) => {
+          return this.onCreate.apply(this, args);
+        }
+      });
+
+      if (!window.OO) {
+        this.loadPlayer();
+      } else {
+        this.createPlayer();
+      }
+    }
+    ```
+
+    Note the usage of `...args` instead of `arguments` keyword. **Why?**
+
+    Well, we used `arguments` and it turned out that **arrow functions don't have lexical `arguments`**. So instead `arguments` from setupPlayer were used. What we really wanted was to use `arguments` passed to onCreate hook called by Ooyala, hence usage of [rest parameter](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/rest_parameters).
+
+Recommended read:
+* http://blog.getify.com/arrow-this/
+* http://toddmotto.com/es6-arrow-functions-syntaxes-and-lexical-scoping/
+* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions
+
+### Default Exports
+
+**Always add export default**
+
+When you don't know which property you should export as default a one think about the property which is the best representative of particular module or which one is going to be used externally most often.
+
+### Classes VS Modules
+
+**Favor Modules over Classes.**
+
+However, the question remains, when I should use class?  
+When it comes to answer to this question I'm too biased to answer that's why I recommend this two articles:
+
+*   [i-would-never-argue-that-es6-classes-are-clearly-a-better-choice-than-composition-modules-or-prototypal-OO](https://medium.com/@rauschma/i-would-never-argue-that-es6-classes-are-clearly-a-better-choice-than-composition-modules-or-891e462da85b#.bfhfjc74j)
+*   [a-simple-challenge-to-classical-inheritance-fans](https://medium.com/javascript-scene/a-simple-challenge-to-classical-inheritance-fans-e78c2cf5eead#.vh341a5iq)[](https://medium.com/javascript-scene/a-simple-challenge-to-classical-inheritance-fans-e78c2cf5eead#.vh341a5iq)
+
+### Imports first
+
+**Imports should be the first thing you put in your files.**  
+
+_Note: ES6 imports are declarative and meant for static analysis. They cannot be dynamic._  
+
+## ESLint
+
+### How to read .eslint.rc
+
+Here is small piece of our .eslint.rc file:
+```json
+"no-process-exit": 0,
+"no-restricted-modules": 0,
+"no-sync": 0,
+"array-bracket-spacing": [2, "never"]
+```
+What all those things mean?

```

Comment:
Maybe we should add a link to http://eslint.org/docs/rules/ so it's easier to understand what those rules mean.